### ISSUE TITLE
Update pyzmq dependency to latest

### ISSUE THREAD

**Issue by phillxnet** [rockstor/rockstor-core#2746]
Our current explicit pinning was due to a Py2.7 dependency, however we have now exceeded the Py3.6 stated newest dependency for this pinning:

> # pyzmq requires libzmq5 on system unless in wheel form.
> pyzmq = "==19.0.2"  # Last specifying Python 2 on PyPi page.

(19.0.2 released July 2020)

It is proposed that we move to the latest version to remove blocker for ongoing Python and Django updates:

PyPi: https://pypi.org/project/pyzmq/
Project page: https://pyzmq.readthedocs.io/en/latest/
> Binary distributions (wheels on PyPI) of PyZMQ ship with the stable version of libzmq at the time of release, ...

Changelog: https://pyzmq.readthedocs.io/en/latest/changelog.html

Latest: (25.1.1 released August 2023)

**Comment by phillxnet** [comment]
Summary of major changes versions:
https://pyzmq.readthedocs.io/en/latest/changelog.html#id26
>20.0 is a major version bump because of dropped support for old Pythons and some changes in packaging, but there are only small changes for users with relatively recent versions of Python.

https://pyzmq.readthedocs.io/en/latest/changelog.html#id25
> pyzmq 21 is a major version bump because of dropped support for old Pythons and some changes in packaging. CPython users should not face major compatibility issues if installation works at all :) PyPy users may see issues with the new implementation of send/recv.

https://pyzmq.readthedocs.io/en/latest/changelog.html#id14
> - Compatibility fixes: Update to Cython 0.29.30 (required for Python 3.11 compatibility)
> - New wheels for PyPy 3.9

https://pyzmq.readthedocs.io/en/latest/changelog.html#id11
23.2.1
> - First release with wheels for Python 3.11 (thanks cibuildwheel!).
> - linux aarch64 wheels now bundle the same libzmq (4.3.4) as all other builds, thanks to switching to native arm builds on CircleCI.

https://pyzmq.readthedocs.io/en/latest/changelog.html#id10
> - pyzmq 24 has two breaking changes (one only on Windows), though they are not likely to affect most users.

https://pyzmq.readthedocs.io/en/latest/changelog.html
26
> pyzmq 26 is a small release, but with some big changes nobody should notice. The Cython backend has been rewritten using Cython 3’s pure Python mode.

**Comment by phillxnet** [comment]
Removing our explicit pinning and we have:
```
lbuildvm:/opt/rockstor # poetry update
Updating dependencies
Resolving dependencies... (7.4s)

Writing lock file

Package operations: 0 installs, 1 update, 0 removals

  • Updating pyzmq (19.0.2 -> 25.1.1)
```

**Comment by phillxnet** [comment]
Noting that pyzmq now has an external Python dependency:
```
pyzmq 25.1.1 Python bindings for 0MQ
└── cffi *
    └── pycparser * 
```
with no version specified. This same library is also a versioned depencency in:
```
gevent 23.9.1 Coroutine-based network library
├── cffi >=1.12.2
│   └── pycparser * 
...
```
and
```
django-oauth-toolkit 2.3.0 OAuth2 Provider for Django
├── django >=2.2,<4.0.0 || >4.0.0
│   ├── asgiref >=3.3.2,<4 
│   │   └── typing-extensions >=4 
│   ├── pytz * 
│   └── sqlparse >=0.2.2 
├── jwcrypto >=0.8.0
│   ├── cryptography >=3.4 
│   │   └── cffi >=1.12 
│   │       └── pycparser *
...
```

**Comment by phillxnet** [comment]
Closing as:
Fixed by #2747
